File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/FlavorOfTheMonth.lua
function GetName()
  return "Flavor Of The Month"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TakingDamage }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == TakingDamage then
    local spell = GetCurrentSpell()
    for i = FIR, LGT do
      if spell:MasteriesContain(i) then
        char:BuffStat(i, 0.5)
      end
    end
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/FlavorOfTheMonth.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Afflictions
{
    public class FlavorOfTheMonth : Affliction
    {
        public override string Name { get { return "Flavor Of The Month"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TakingDamage, 
                };
            }
        }

        protected override void OnTakingDamage(ICharacter character, ref double damage)
        {
            var spell = GetCurrentSpell();
            var masteries = spell.Masteries;
            for (var i = Stat.FIR; i <= Stat.LGT; ++i)
            {
                if (masteries.Contains((EleMastery) i))
                {
                    character.BuffStat(i, .5);
                }
            }
        }
    }
}